src/views/Gantt.vue

```diff
@@ -0,0 +1,267 @@
+            label="Select tasks"
+            ref="selectTasks"
+            @update:search="updateSelectionOptions"
+          >
+            <template
+              v-slot:prepend-item
+              v-if="this.showSelectAll"
+            >
+              <v-list-item
+                ripple
+                @click="selectSearchResults"
+              >
+                Select all search results
+              </v-list-item>
+              <v-list-item
+                ripple
+                @click="deselectSearchResults"
+              >
+                Remove all search results
+              </v-list-item>
+            <v-divider class="mt-2"></v-divider>
+            </template>
+          </v-autocomplete>
+        </v-col>
+        <v-col
+          cols="12"
+          md="4"
+          class="mb-2 mb-md-0"
+        >
+          <v-select
+            id="c-gantt-filter-job-timings"
+            :items="timingOptions"
+            prefix="Displaying:"
+            v-model="jobsFilter.timingOption"
+          />
+        </v-col>
+        <v-col
+          cols="12"
+          md="4"
+          class="pl-md-2 mb-2 mb-md-0"
+        >
+          <v-select
+            id="c-gantt-filter-job-platforms"
+            :items="platformOptions"
+            prefix="Platform:"
+            v-model="jobsFilter.platformOption"
+          />
+        </v-col>
+      </v-row>
+      <v-row no-gutters>
+        <v-col
+          cols="12"
+          md="4"
+          class="pr-md-2 mb-2 mb-md-0"
+        >
+          <v-select
+            id="c-gantt-tasks-per-page"
+            :items="$options.taskChoices"
+            prefix="Tasks Per Page"
+            v-model="tasksPerPage"/>
+        </v-col>
+      </v-row>
+      <div
+        id="gantt-toolbar"
+        class="d-flex align-center flex-wrap my-2 col-gap-2 row-gap-4">
+
+        <!-- Toolbar -->
+        <v-defaults-provider>
+
+          <!-- Box plot sort input teleports here -->
+        </v-defaults-provider>
+      </div>
+      <GanttChart
+        :jobs="filteredJobs"
+        :timing-option="timingOption"
+        :tasks-per-page="tasksPerPage"
+      />
+    </v-container>
+  </div>
+</template>
+
+<script>
+import {
+  debounce
+} from 'lodash'
+import gql from 'graphql-tag'
+import { getPageTitle } from '@/utils/index'
+import graphqlMixin from '@/mixins/graphql'
+import GanttChart from '@/components/cylc/gantt/GanttChart.vue'
+import {
+  matchTasks,
+  platformOptions
+} from '@/components/cylc/analysis/filter'
+/** List of fields to request for each job */
+const jobFields = [
+  'name',
+  'id',
+  'submittedTime',
+  'startedTime',
+  'finishedTime',
+  'platform'
+]
+/** The query which retrieves historical Job timing statistics */
+const QUERY = gql`
+query ganttQuery ($workflows: [ID]) {
+  jobs(live: false, workflows: $workflows) {
+    ${jobFields.join('\n')}
+  }
+}
+`
+/** The callback which gets called when data comes in from the query */
+class GanttCallback {
```

Comment:
@oliver-sanders should this be?
```suggestion
class GanttCallback extends DeltasCallback {
```

Reply:
I don't think it makes a functional difference, but it's good practice to inherit from the base class.